packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware/index.ts

```diff
@@ -0,0 +1,485 @@
+              output_currency: outputCurrency,
+              output_type: outputType,
+              type: AnalyticsType.EVENT
+            })
+
+            break
+          }
+          default: {
+            break
+          }
+        }
+        break
+      }
+      case AT.components.swap.HANDLE_SWAP_MAX_AMOUNT_CLICK: {
+        const state = store.getState()
+        const inputCurrency = state.form.initSwap.values.BASE.coin
+        const inputType =
+          state.form.initSwap.values.BASE.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+        const outputCurrency = state.form.initSwap.values.COUNTER.coin
+        const outputType =
+          state.form.initSwap.values.COUNTER.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+
+        analytics.push(AnalyticsKey.SWAP_AMOUNT_MAX_CLICKED, {
+          id,
+          input_currency: inputCurrency,
+          input_type: inputType,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+          output_currency: outputCurrency,
+          output_type: outputType,
+          type: AnalyticsType.EVENT
+        })
+
+        break
+      }
+      case AT.components.swap.HANDLE_SWAP_MIN_AMOUNT_CLICK: {
+        const state = store.getState()
+        const inputCurrency = state.form.initSwap.values.BASE.coin
+        const inputType =
+          state.form.initSwap.values.BASE.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+        const outputCurrency = state.form.initSwap.values.COUNTER.coin
+        const outputType =
+          state.form.initSwap.values.COUNTER.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+
+        analytics.push(AnalyticsKey.SWAP_AMOUNT_MIN_CLICKED, {
+          id,
+          input_currency: inputCurrency,
+          input_type: inputType,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+          output_currency: outputCurrency,
+          output_type: outputType,
+          type: AnalyticsType.EVENT
+        })
+        break
+      }
+      case AT.components.swap.CHANGE_BASE: {
+        const inputCurrency = action.payload.account.coin
+        const inputType = action.payload.account.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+
+        analytics.push(AnalyticsKey.SWAP_FROM_SELECTED, {
+          id,
+          input_currency: inputCurrency,
+          input_type: inputType,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+
+          type: AnalyticsType.EVENT
+        })
+        break
+      }
+      case AT.components.swap.CHANGE_COUNTER: {
+        const inputCurrency = action.payload.account.coin
+        const inputType = action.payload.account.type === 'CUSTODIAL' ? 'TRADING' : 'USERKEY'
+
+        analytics.push(AnalyticsKey.SWAP_RECEIVE_SELECTED, {
+          id,
+          input_currency: inputCurrency,
+          input_type: inputType,
+          nabuId,
+          originalTimestamp: getOriginalTimestamp(),
+
+          type: AnalyticsType.EVENT
+        })
+        break
```

Comment:
this case is the exact same as the case above. why do we need both?

Reply:
Because it actually has the different keys, meaning that it's easier to track when we change the base or the counter